cellmlmanip/units.py

```diff
@@ -162,146 +143,145 @@ def __init__(self, expression):
 
 
 class UnitStore(object):
-    """Wraps the underlying Pint UnitRegistry to provide unit handling for the model's Sympy expressions.
+    """
+    Maps string names to ``pint.Unit`` and ``pint.Quantity`` objects (providing a unit namespace), and provides unit
+    checking and conversion.
+
+    Each ``UnitStore`` object contains the units defined in the CellML specification, and more units can be added by the
+    user.
+
+    By default, each ``UnitStore`` maintains its own internal ``pint.UnitRegistry``, so that units from different stores
+    cannot be compared or interact with each other. To allow comparison and conversion between unit stores, an existing
+    ``UnitStore`` can be passed in at construction time, so that the underlying registry will be shared.
+
+    For example::
+
+        a = UnitStore()
+        b = UnitStore()
+
+    creates two separated unit stores, each with their own namespace, and without the option to compare units from ``a``
+    to units from ``b``.
+
+    But::
 
-    Getting and checking units is handled by this class.
+        a = UnitStore()
+        b = UnitStore(a)
+        c = UnitStore(b)
+
+    creates three unit stores, each with their own namespace, but with a single underlying registry for all three
+    stores, allowing e.g. units from ``c`` to be converted to units from ``a``.
+
+    :param store: An existing ``UnitStore`` to share a unit registry with.
     """
+    _next_id = 0
 
-    def __init__(self, model):
-        """Initialise a UnitStore instance; wraps the unit registry and handles addition of new
-        unit definitions"""
-        cellml_unit_definition = os.path.join(
-            os.path.dirname(__file__), 'data', 'cellml_units.txt')
-        self.ureg = pint.UnitRegistry(cellml_unit_definition)
+    def __init__(self, store=None):
 
-        # units that are defined and added to the unit registry, on top of default cellml units
-        self.custom_defined = set()
+        # Assign this unit store a unique id, to keep its namespace separated from any other unit stores sharing the
+        # same registry.
+        self._id = 'store' + str(UnitStore._next_id) + '_'
+        UnitStore._next_id += 1
 
-        # Keep reference to the underlying model
-        self.model = model
+        # Get unit registry
+        if store is None:
+            # Create new unit registry, configured for CellML units.
+            self._registry = pint.UnitRegistry(os.path.join(os.path.dirname(__file__), 'data', 'cellml_units.txt'))
+        else:
+            # Share a registry and calculator with the given unit store
+            self._registry = store._registry
 
-        # Unit calculator
-        self.calculator = UnitCalculator(self.ureg)
+        # User defined units
```

Comment:
I'd note here that the set just stores names. I thought initially it contained the objects.